### PR TITLE
docs: Add docstrings to Seam class

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
wrt https://github.com/seamapi/python/issues/86

### CODE REVIEW

File: seam/seam.py [hunk 1]
```diff
@@ -10,7 +10,38 @@
...
+        devices (Devices): Interface for device-related operations.
+        events (SeamEvents): Interface for event-related operations.
+        locks (Locks): Interface for lock-related operations.
+        networks (Networks): Interface for network-related operations.
+        noise_sensors (NoiseSensors): Interface for noise sensor operations.
+        phones (Phones): Interface for phone-related operations.
+        thermostats (Thermostats): Interface for thermostat operations.
+        user_identities (UserIdentities): Interface for user identity operations.
+        webhooks (Webhooks): Interface for webhook operations.
+        workspaces (Workspaces): Interface for workspace operations.
```
Comment (razor-x) [inline seam/seam.py]: If we are going to document this, we need to make sure it's part of codegen. But do we need this here? Wouldn't' the docs for these attributes be at the attribute / method level?

Comment (andrii-balitskyi) [reply seam/seam.py]: I wasn't sure about adding these so happy to move them to attribute / method level

Comment (andrii-balitskyi) [reply seam/seam.py]: a9109042009db96f2351ccaf21bc3ab5dea67ad1

File: seam/seam.py [hunk 2]
```diff
@@ -25,19 +40,45 @@ def __init__(
...
+        :raises SeamHttpUnauthorizedError: When the provided authentication
+            credentials (api_key or personal_access_token) are invalid
+        :raises SeamHttpInvalidInputError: When the API request contains
+            invalid input data
+        :raises SeamActionAttemptFailedError: When an action attempt fails to
+            complete successfully (only when wait_for_action_attempt is
+            enabled)
+        :raises SeamActionAttemptTimeoutError: When an action attempt exceeds
+            the specified timeout duration (only when wait_for_action_attempt
+            is enabled)
```
Comment (razor-x) [inline seam/seam.py]: These are all raised by the individual methods. Does it make sense to put them on the constructor? I wonder what other libs do, since often errors are raised in many places.

File: seam/seam.py [hunk 3]
```diff
@@ -48,11 +89,11 @@ def __init__(
             workspace_id=workspace_id,
             endpoint=endpoint,
         )
-        defaults = {"wait_for_action_attempt": wait_for_action_attempt}
+        self.defaults = {"wait_for_action_attempt": wait_for_action_attempt}
```
Comment (razor-x) [inline seam/seam.py:84]: Why is this added to self now? Do we need it, and if so, maybe it should be private?

File: seam/seam.py (hunk 2 above)
Comment (andrii-balitskyi) [reply seam/seam.py]: Oh yeah I meant to ask about this but forgot. Probably it doesn't make sense to put them all here. Http errors can go to `SeamHttpClient` docstring and action attempt ones to the `ActionAttempts` route class docstring. I think `SeamInvalidOptionsError` and `SeamInvalidTokenError` is fine to leave here.

> I wonder what other libs do, since often errors are raised in many places.

I briefly looked through a couple of libs and it seems that mostly exceptions are documented where they're raised. I also found instances where exception docstring is also propagated up where the method throwing it is used like [here](https://github.com/sqlalchemy/sqlalchemy/blob/96f1172812f858fead45cdc7874abac76f45b339/lib/sqlalchemy/orm/base.py#L384-L424).

Even though `InvalidOptionsError` is raised in `get_auth_headers` which in turn is called in `parse_options` that we use in the Seam constructor, I think it still makes sense to document it on the constructor because none of those functions handle the error and it'll get raised when the constructor is called. Plus if the error is not documented on the constructor, it won't be clear from the IDE intellisense that when calling Seam constructor such errors can be raised. Ideally we should document the error on each level where it can be raised?

File: seam/seam.py (hunk 3 above)
Comment (andrii-balitskyi) [reply seam/seam.py:84]: Added this because I double checked the [JS SDK](https://github.com/seamapi/javascript-http/blob/295dc1707b8e6e1fdf82d53311e0f152ac74a11b/src/lib/seam/connect/seam-http.ts#L53) and it assigns `defaults` as an instance variable. Hence the change.

**Comment by razor-x** [comment]
@DebbieAtSeam Do you want to review the language we use in the docstrings here? See also https://github.com/seamapi/python/pull/113/files#diff-6aa596a1d9b2ede6d6729f736e0d98900fee277a5b3f129509f9485f4ba86d72

**Comment by DebbieAtSeam** [comment]
@razor-x Yes, I'd like to do a light copy edit, please. Thanks!

**Comment by DebbieAtSeam** [comment]
@razor-x Further, are the items with new docstrings here things that do not exist (and could not be documented once) in a centralized location that could populate to all the SDKs?

**Comment by razor-x** [comment]
@DebbieAtSeam 

> Further, are the items with new docstrings here things that do not exist (and could not be documented once) in a centralized location that could populate to all the SDKs?

There will be similarity across SDKs but it will be language specific. I'd say we should limit the content as much as possible and link to the docs if we need to expand. So, even this sentence feels mostly extraneous 

> This class provides methods to authenticate and interact with various
>     Seam API endpoints,
>     including devices, access codes, action_attempts, and more. It supports authentication via API key or personal access token.

**Comment by andrii-balitskyi** [comment]
@DebbieAtSeam @razor-x, are we good to merge this? Or the docstring language is still to be reviewed?

**Comment by DebbieAtSeam** [comment]
Let's merge it now and then think about editing it and syncing the wording with the JS version later on.

**Review by razor-x — APPROVED** [review]
